src/tool/xmeta/cpp/models/class_model.h

```diff
@@ -5,32 +5,41 @@
 #include <string>
 #include <vector>
 #include <variant>
-
-#include "event_model.h"
-#include "interface_model.h"
-#include "method_model.h"
-#include "model_ref.h"
 #include "class_or_interface_model.h"
-#include "property_model.h"
+#include "model_types.h"
 
 namespace xlang::xmeta
 {
-    // Note: All classes are implicitly public, and explicitly specifying this is not allowed.
-    enum class class_semantics
+    struct class_semantics
```

Comment:
You might be overusing the term semantics. It's not clear (from the name) the difference between type_semantics, class_semantics and class_type_semantics

Reply:
I've change this to class_modifier